pkg/virt-launcher/virtwrap/manager_test.go

```diff
@@ -1622,6 +1624,95 @@ var _ = Describe("Manager", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(domSpec).ToNot(BeNil())
 		})
+
+		Context("on call to GetGuestOSInfo", func() {
+			var libvirtmanager DomainManager
+			var agentStore agentpoller.AsyncAgentStore
+
+			BeforeEach(func() {
+				agentStore = agentpoller.NewAsyncAgentStore()
+				libvirtmanager, _ = NewLibvirtDomainManager(mockConn, testVirtShareDir, nil, 0, &agentStore, "/usr/share/OVMF")
+			})
+
+			It("should return nil when no osInfo exists in the cache", func() {
+				osInfo := libvirtmanager.GetGuestOSInfo()
+				Expect(osInfo).To(BeNil())
+			})
+
+			It("should return osInfo when osInfo exists in the cache", func() {
+				fakeInfo := api.GuestOSInfo{
+					Name: "TestGuestOSName",
+				}
+
+				agentStore.Store(agentpoller.GET_OSINFO, fakeInfo)
+				osInfo := libvirtmanager.GetGuestOSInfo()
+				Expect(reflect.DeepEqual(&fakeInfo, osInfo)).To(BeTrue())
+			})
+		})
+
+		Context("on call to InterfacesStatus", func() {
+			var libvirtmanager DomainManager
+			var agentStore agentpoller.AsyncAgentStore
+			domInterfaces := []api.Interface{
+				{
+					MAC: &api.MAC{
+						MAC: "00:00:00:00:00:01",
+					},
+					Alias: api.NewUserDefinedAlias("eth1"),
+				},
+			}
+			fakeInterfaces := []api.InterfaceStatus{
+				{
+					Name: "eth2",
+					Mac:  "00:00:00:00:00:02",
+				},
+			}
+
+			BeforeEach(func() {
+				agentStore = agentpoller.NewAsyncAgentStore()
+				libvirtmanager, _ = NewLibvirtDomainManager(mockConn, testVirtShareDir, nil, 0, &agentStore, "/usr/share/OVMF")
+			})
+
+			It("should return nil when no interfaces exists in the cache, nor as argument", func() {
+				interfaces := libvirtmanager.InterfacesStatus(nil)
+				Expect(interfaces).To(BeNil())
+			})
+
+			It("should return nil when no interfaces exists in the cache", func() {
+				interfaces := libvirtmanager.InterfacesStatus(domInterfaces)
+				Expect(interfaces).To(BeNil())
```

Comment:
can be inline

Reply:
less reable imo,
personally its much easier to read in two lines, what we get and what we do with it
but ok, already did.. (on next push)